server/src/main/java/org/elasticsearch/plugins/PluginBundle.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.plugins;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.LinkedHashSet;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * A "bundle" is a group of jars that will be loaded in a single classloader
```

Comment:
This isn't entirely true: spi jars may be loaded in a separate loader.

Reply:
I changed it to "A "bundle" is a group of jars for an Elasticsearch plugin. Ultimately, the PluginsService will load these jars, creating any necessary classloaders."